src/simplewallet/simplewallet.cpp

```diff
@@ -465,6 +476,595 @@ bool simple_wallet::print_fee_info(const std::vector<std::string> &args/* = std:
+
+  const std::string filename = args[0];
+  try
+  {
+    std::vector<tools::wallet2::multisig_info> outs = m_wallet->export_multisig();
+
+    std::stringstream oss;
+    boost::archive::portable_binary_oarchive ar(oss);
+    ar << outs;
+
+    std::string magic(MULTISIG_EXPORT_FILE_MAGIC, strlen(MULTISIG_EXPORT_FILE_MAGIC));
+    const cryptonote::account_public_address &keys = m_wallet->get_account().get_keys().m_account_address;
+    std::string header;
+    header += std::string((const char *)&keys.m_spend_public_key, sizeof(crypto::public_key));
+    header += std::string((const char *)&keys.m_view_public_key, sizeof(crypto::public_key));
+    crypto::public_key signer = m_wallet->get_multisig_signer_public_key();
+    header += std::string((const char *)&signer, sizeof(crypto::public_key));
+    std::string ciphertext = m_wallet->encrypt_with_view_secret_key(header + oss.str());
+    bool r = epee::file_io_utils::save_string_to_file(filename, magic + ciphertext);
+    if (!r)
+    {
+      fail_msg_writer() << tr("failed to save file ") << filename;
+      return true;
+    }
+  }
+  catch (const std::exception &e)
+  {
+    LOG_ERROR("Error exporting multisig info: " << e.what());
+    fail_msg_writer() << tr("Error exporting multisig info: ") << e.what();
+    return true;
+  }
+
+  success_msg_writer() << tr("Multisig info exported to ") << filename;
+  return true;
+}
+
+bool simple_wallet::import_multisig(const std::vector<std::string> &args)
+{
+  bool ready;
+  uint32_t threshold, total;
+  if (!m_wallet->multisig(&ready, &threshold, &total))
+  {
+    fail_msg_writer() << tr("This wallet is not multisig, or not yet finalized");
+    return true;
+  }
+  if (args.size() < threshold - 1)
+  {
+    fail_msg_writer() << tr("usage: import_multisig <filename1> [<filename2>...] - one for each other participant");
+    return true;
+  }
+  if (m_wallet->ask_password() && !get_and_verify_password())
+    return true;
+
+  std::vector<std::vector<tools::wallet2::multisig_info>> info;
+  std::unordered_set<crypto::public_key> seen;
+  for (size_t n = 0; n < args.size(); ++n)
+  {
+    const std::string filename = args[n];
+    std::string data;
+    bool r = epee::file_io_utils::load_file_to_string(filename, data);
+    if (!r)
+    {
+      fail_msg_writer() << tr("failed to read file ") << filename;
+      return true;
+    }
+    const size_t magiclen = strlen(MULTISIG_EXPORT_FILE_MAGIC);
+    if (data.size() < magiclen || memcmp(data.data(), MULTISIG_EXPORT_FILE_MAGIC, magiclen))
+    {
+      fail_msg_writer() << tr("Bad multisig info file magic in ") << filename;
+      return true;
+    }
+
+    try
+    {
+      data = m_wallet->decrypt_with_view_secret_key(std::string(data, magiclen));
+    }
+    catch (const std::exception &e)
+    {
+      fail_msg_writer() << tr("Failed to decrypt ") << filename << ": " << e.what();
+      return true;
+    }
+
+    const size_t headerlen = 2 * sizeof(crypto::public_key) + sizeof(crypto::hash);
```

Comment:
Same comment again: this should be `3 * sizeof(crypto::public_key)` since `signer` is declared as `public_key` instead of `hash`.